src/instructlab/model/list.py

```diff
@@ -80,72 +87,36 @@ def print_table(headers: list[str], data: list[list[str]]):
     print(joining_line)
 
 
-AnalyzeResultGGUF = list[str]
-AnalyzeResultDir = list[list[str]]
+def get_model_info(path: str, mtime: float, n_bytes: int) -> ModelInfo:
+    return (
+        path,
+        time.strftime("%Y-%m-%d %H:%M:%S", time.localtime(mtime)),
+        get_bytes_size_string(n_bytes),
+    )
 
 
-def _analyze_gguf(entry: Path) -> AnalyzeResultGGUF:
-    # stat the gguf, add it to the table
-    stat = Path(entry.absolute()).stat(follow_symlinks=False)
-    f_size = stat.st_size
-    f_size, magnitude = convert_bytes_to_proper_mag(f_size)
-    # add to table
-    modification_time = os.path.getmtime(entry.absolute())
-    modification_time_readable = time.strftime(
-        "%Y-%m-%d %H:%M:%S", time.localtime(modification_time)
-    )
-    return [entry.name, modification_time_readable, f"{f_size:.1f} {magnitude}"]
+def _analyze_gguf(entry: Path) -> ModelInfo:
+    stat = entry.stat(follow_symlinks=False)
+    return get_model_info(entry.name, stat.st_mtime, stat.st_size)
 
 
 def _analyze_dir(
     entry: Path, list_checkpoints: bool, directory: str
-) -> AnalyzeResultDir:
-    actual_model_name = ""
-    all_files_sizes = 0
-    add_model = False
-    models = []
-    # walk entire dir.
-    for root, _, files in os.walk(entry.as_posix()):
-        normalized_path = os.path.normpath(root)
-        # Split the path into its components
-        parts = normalized_path.split(os.sep)
-        # Get the last two parts and join them back into a path
-        last_two_parts = os.path.join(parts[-2], parts[-1])
-        # if this is a dir it could be:
```

Comment:
can we please preserve these comments, they are useful

Reply:
OK I will rewrite them a bit then.

Reply:
I've updated as per my understanding of the code, but feel free to correct me. I think either `gguf` cases are not applicable, or the code has a bug. You decide what the intent was.

Reply:
I think this PR may actually fix the gguf handling: https://github.com/instructlab/instructlab/pull/1895 So I guess it was intended, but broken before?